### DISCUSSION TITLE
Adding my new membership inference attack

### DISCUSSION THREAD

**Discussion by giladcohen** [Trusted-AI/adversarial-robustness-toolbox#1721] [Q&A] [answered]
Hello,
I have developed a new membership inference attack that will be published soon.
I want to add my attack to:
[adversarial-robustness-toolbox/art/attacks/inference/membership_inference](https://github.com/Trusted-AI/adversarial-robustness-toolbox/tree/main/art/attacks/inference/membership_inference)

However, my attack makes use of influence functions and thus requires including routines from [this repo](https://github.com/giladcohen/influence_functions).

I have implemented it successfully in [my private repo](https://github.com/giladcohen/sif_mi_attack/blob/master/art/attacks/inference/membership_inference/self_influence_function_attack.py), but I am not sure this is "legal" to do in the official ART repo. More precisely, see lines #15-17 in my code. I import the `pytorch_influence_functions` module. How can I rely on this module when writing my attack to the official ART repo?

Thanks!

**Comment by beat-buesser** [comment, marked as answer]
Hi @giladcohen Thank you for your interest in ART. This sounds interesting.

ART has a lot of dependencies that are required only by one or a few tools. In such cases we treat the dependency as optional and we suggest that the tool checks if the dependency is available and if not raises and exception with an explanation of how to install it.

I would be very interested to read you paper, when are you planning to publish it?

**Reply by giladcohen** [reply]
Can you please refer me to an example attack that implements an optional dependency in ART?
My paper was sent for review to NeurIPS 2022 and it will be published on Arxiv in a few days.

**Reply by beat-buesser** [reply]
Great, please send me a link to the paper once it's published.

This could be an example for the dependency check: https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/b0d86c4489f92275d828e7cde7e9d0a34181f2fe/art/estimators/certification/__init__.py#L7
